src/main/java/com/three_tech_solutions/slot_app/components/payment_processors/PaymentProcessor.java

```diff
@@ -14,17 +14,22 @@
 
 @Slf4j
 public abstract class PaymentProcessor {
-    public Payment createStudentPayment(Student student, int newPaymentNumber){
-        log.info("Creando pago para el estudiante: {}", student);
-        log.info("El plan es {}", getCurrentPlan());
+    protected Payment createPayment(Student student, int newPaymentNumber, double amount) {
         return new Payment(
-                getPaymentAmount(student),
+                amount,
                 PaymentStatus.EN_TERMINO,
                 getExpirationDate(student),
                 newPaymentNumber,
                 student
         );
     }
+
+    public Payment createStudentPayment(Student student, int newPaymentNumber){
+        log.info("Creando pago para el estudiante: {}", student);
+        log.info("El plan es {}", getCurrentPlan());
+        return createPayment(student, newPaymentNumber, getPaymentAmount(student));
```

Comment:
Acá ahora, tendriamos que decirle cuál es la fecha, y así como la buscábamos antes en el createPayment, podemos hacerlo también en este método

```suggestion
        return createPayment(getExpirationDate(student), student, newPaymentNumber, getPaymentAmount(student));

```